server.js

```diff
@@ -1,37 +1,125 @@
+import dotenv from 'dotenv'
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+dotenv.config()
 
+
+import bootcamps from './data/bootcamps.json'
+
+//these 3 lines are responsible for connecting the server to the database
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
+//To create the schema with data types
+const bootcampSchema = new mongoose.Schema({
+  name: String,
+  image: String,
+  review: String,
+  courses: String,
+  subjects: String,
+  description: String,
+  cost: String,
+  duration: String,
+  isFlexible: Boolean,
+  guaranteesJob: Boolean ,
+  hasGenderPerspective: Boolean,
+  url: String  
+})
+
+//To create the Bootcamp model
+const Bootcamp = mongoose.model('Bootcamp', bootcampSchema)
+
+//Seeding the database. Will only run if
+// RESET_DB environment variable is present and is true
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+   await Bootcamp.deleteMany()
+
+//Creates a new entry for each bootcamp
+  await bootcamps.forEach(item => {
+    const newBootcamp = new Bootcamp(item)
+    newBootcamp.save()
+  })
+}
+seedDatabase()
+}
+
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
+
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
   res.send('Hello world')
 })
+//The endpoint to get all the coding bootcamps in the list or filter by guaranteed job, gender perspective or id
+app.get('/bootcamps', async (req, res) => {
+  const { name, guaranteesJob, hasGenderPerspective } = req.query
+
+  if (name) {
+    const bootcamps = await Bootcamp.find({
+      name: {
+        $regex: new RegExp(name, "i")}
+    })
+    res.json(bootcamps)
+  }
+
+  if (guaranteesJob) {
+    const bootcamps = await Bootcamp.find({
+      guaranteesJob:true
+    })
+    res.json(bootcamps)
+  }
+
+  if (hasGenderPerspective) {
+    const bootcamps = await Bootcamp.find({
+      hasGenderPerspective: true
+    })
+    res.json(bootcamps)
+  }
```

Comment:
I can image how useful these query parameters would be on the frontend side. Good job!✨